### DISCUSSION TITLE
Issues Installing 'fairseq' with CUDA 12.4 Support

### DISCUSSION THREAD

**Discussion by royla81** [community/community#120917] [Programming Help]
### Body

Hi there,

I've been attempting to install the `fairseq` library on a Windows 10 system with NVIDIA CUDA 12.4 support, but I am encountering persistent errors. Below are the details of my environment and the steps I have taken so far:

- Operating System: Windows 10 (64-bit)
- Python version: Python 3.10 (managed with Anaconda)
- CUDA version: 12.4 (as confirmed by `nvcc --version`)
- GPU: GeForce RTX 2070 SUPER
- fairseq version: Attempting to install the latest from the GitHub repository

Despite following the installation instructions, I am faced with a recurring error stating that CUDA is not found on the system, and that I should set the CUDA_HOME or CUDA_PATH environment variables or add NVCC to my PATH. Here is the error message:

RuntimeError:
CUDA was not found on the system, please set the CUDA_HOME or the CUDA_PATH
environment variable or add NVCC to your system PATH. The extension compilation will fail.

I have ensured that the CUDA Toolkit is correctly installed, NVCC is accessible, and the appropriate environment variables have been set (both temporarily in the session and permanently in the system settings). Additionally, I have tried running the Anaconda Prompt as an administrator to ensure sufficient privileges.

Here is a brief summary of the commands executed and their sequence:

1. Conda environment creation and activation with Python 3.10
2. Clone the fairseq repository
3. Run `pip install --editable .` within the fairseq directory

Unfortunately, this results in the aforementioned error. I would greatly appreciate any advice or suggestions on how to resolve this issue. Perhaps there is a specific version of `fairseq` that is known to work well with CUDA 12.4, or there might be some additional steps that I need to take to ensure compatibility.

Thank you for your assistance,
Sincerely,
Roy

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ebndev** [comment]
Hey @royla81! 

Were you able to figure this out? It doesn't look like we have that many posts in the Community about NVIDIA but it would be great to reference this post in the future if someone else encounters an issue.
